main.go

```diff
@@ -0,0 +1,96 @@
+///////////////////////////////////////////////////////////////////////
+// Copyright (c) 2018 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+///////////////////////////////////////////////////////////////////////
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"strconv"
+
+	"github.com/dispatchframework/funky/pkg/funky"
+)
+
+const (
+	serversEnvVar   = "SERVERS"
+	serverCmdEnvVar = "SERVER_CMD"
+)
+
+type funkyHandler struct {
+	router funky.Router
+}
+
+func (f funkyHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	reader := r.Body
+
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(reader)
+	rawBody := buf.Bytes()
+	var body map[string]interface{}
+	err := json.Unmarshal(rawBody, &body)
+	if err != nil {
+		resp := funky.Response{
+			Context: funky.Context{
+				Error: funky.Error{
+					ErrorType: funky.InputError,
+					Message:   "Invalid Input",
+				},
+			},
+		}
+		out, _ := json.Marshal(resp)
+		fmt.Fprintf(w, string(out))
+		return
+	}
+
+	resp, _ := f.router.Delegate(body)
+
+	out, _ := json.Marshal(resp)
+
+	fmt.Fprintf(w, string(out))
+}
+
+func main() {
+	numServers, err := strconv.Atoi(os.Getenv(serversEnvVar))
+	if err != nil {
+		log.Fatalf("Unable to parse %s environment variable", serversEnvVar)
+	}
+	if numServers < 1 {
+		numServers = 1
+	}
+
+	serverCmd := os.Getenv(serverCmdEnvVar)
+	serverFactory, err := funky.NewDefaultServerFactory(serverCmd)
+	if err != nil {
+		log.Fatal("Too few arguments to server command.")
+	}
+
+	router, err := funky.NewRouter(numServers, serverFactory)
+	if err != nil {
+		log.Fatalf("Failed creating new router: %+v", err)
+	}
+	handler := funkyHandler{
+		router: router,
+	}
+	server := &http.Server{
+		Addr:    ":8080",
+		Handler: handler,
+	}
+
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt)
+	go func() {
+		<-c
+		router.Shutdown()
+		server.Shutdown(context.TODO())
```

Comment:
This line should go before `router.Shutdown()`: 

`server.Shutdown(ctx)` "gracefully shuts down the server without interrupting any active connections. Shutdown works by first closing all open listeners, then closing all idle connections, and then waiting indefinitely for connections to return to idle and then shut down"